### PR TITLE
*: log format for schedule, scheduler, syncer, api

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Format the log according to our RFC.

### What is changed and how it works?
This PR is going to do the log format for `schedule`, `scheduler`, `syncer`, and `api` packages.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by disksing** [comment]
Does it need to be merged after #1422?

File: server/schedulers/balance_region.go [hunk 1]
```diff
@@ -155,16 +157,16 @@ func (s *balanceRegionScheduler) transferPeer(cluster schedule.Cluster, region *
 	}
 
 	target := cluster.GetStore(storeID)
-	log.Debugf("[region %d] source store id is %v, target store id is %v", region.GetID(), source.GetID(), target.GetID())
+	log.L().Debug("", zap.Uint64("region", region.GetID()), zap.Uint64("source-store", source.GetID()), zap.Uint64("target-store", target.GetID()))
```
Comment (disksing) [inline server/schedulers/balance_region.go]: It seems that we need to unify the naming among all PRs. Here the kebab style is used, but in #1422 space-separated strings are used, see https://github.com/pingcap/pd/pull/1422/files#diff-44da3d359afd2823a31e4fb048bdfdf4R58
/cc @nolouch

File: server/schedulers/balance_leader.go [hunk 2]
```diff
@@ -17,10 +17,12 @@ import (
 	"fmt"
 	"strconv"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/schedulers/balance_leader.go]: Remove empty line?

File: server/schedulers/adjacent_region.go [hunk 3]
```diff
@@ -18,10 +18,12 @@ import (
 	"strconv"
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/schedulers/adjacent_region.go]: Remove empty line?

File: server/schedule/scheduler.go [hunk 4]
```diff
@@ -16,10 +16,12 @@ package schedule
 import (
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/schedule/scheduler.go]: Remove empty line?

File: server/schedule/replica_checker.go [hunk 5]
```diff
@@ -16,10 +16,12 @@ package schedule
 import (
 	"fmt"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/schedule/replica_checker.go]: Remove empty line?

File: server/api/redirector.go [hunk 6]
```diff
@@ -14,13 +14,16 @@
...
 
 import (
+	"fmt"
 	"io/ioutil"
 	"net/http"
 	"net/url"
 	"strings"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/api/redirector.go]: Remove empty line?

File: server/region_syncer/client.go [hunk 7]
```diff
@@ -18,9 +18,11 @@ import (
 	"net/url"
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/region_syncer/client.go]: Remove empty line?

File: server/region_syncer/history_buffer.go [hunk 8]
```diff
@@ -17,8 +17,10 @@ import (
 	"strconv"
 	"sync"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/region_syncer/history_buffer.go]: Remove empty line?

File: server/region_syncer/server.go [hunk 9]
```diff
@@ -19,12 +19,14 @@ import (
 	"sync"
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/region_syncer/server.go]: Remove empty line?

File: server/schedule/merge_checker.go [hunk 10]
```diff
@@ -16,10 +16,12 @@ package schedule
 import (
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/schedule/merge_checker.go]: Remove empty line?

File: server/schedule/operator.go [hunk 11]
```diff
@@ -21,9 +21,11 @@ import (
 	"sync/atomic"
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/schedule/operator.go]: Remove empty line?

File: server/schedule/operator.go [hunk 12]
```diff
@@ -82,7 +84,7 @@ func (ap AddPeer) String() string {
 func (ap AddPeer) IsFinish(region *core.RegionInfo) bool {
 	if p := region.GetStoreVoter(ap.ToStore); p != nil {
 		if p.GetId() != ap.PeerID {
-			log.Warnf("expect %v, but obtain voter %v", ap.String(), p.GetId())
+			log.L().Warn("", zap.String("expect", ap.String()), zap.Uint64("obtain voter", p.GetId()))
```
Comment (disksing) [inline server/schedule/operator.go]: Add some  warning message here.

File: server/schedule/operator.go [hunk 13]
```diff
@@ -111,7 +113,7 @@ func (al AddLearner) String() string {
 func (al AddLearner) IsFinish(region *core.RegionInfo) bool {
 	if p := region.GetStoreLearner(al.ToStore); p != nil {
 		if p.GetId() != al.PeerID {
-			log.Warnf("expect %v, but obtain learner %v", al.String(), p.GetId())
+			log.L().Warn("", zap.String("expect", al.String()), zap.Uint64("obtain learner", p.GetId()))
```
Comment (disksing) [inline server/schedule/operator.go]: Add some warning message.

File: server/schedule/operator.go [hunk 14]
```diff
@@ -140,7 +142,7 @@ func (pl PromoteLearner) String() string {
 func (pl PromoteLearner) IsFinish(region *core.RegionInfo) bool {
 	if p := region.GetStoreVoter(pl.ToStore); p != nil {
 		if p.GetId() != pl.PeerID {
-			log.Warnf("expect %v, but obtain voter %v", pl.String(), p.GetId())
+			log.L().Warn("", zap.String("expect", pl.String()), zap.Uint64("obtain voter", p.GetId()))
```
Comment (disksing) [inline server/schedule/operator.go]: add some warning message

File: server/schedule/operator_controller.go [hunk 15]
```diff
@@ -18,11 +18,13 @@ import (
 	"sync"
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/schedule/operator_controller.go]: Remove empty line?

File: server/schedule/replica_checker.go [hunk 16]
```diff
@@ -224,7 +226,7 @@ func (r *ReplicaChecker) checkBestReplacement(region *core.RegionInfo) *Operator
 	}
 	// Make sure the new peer is better than the old peer.
 	if newScore <= oldScore {
-		log.Debugf("[region %d] newScore %f is not better than oldScore %f", region.GetID(), newScore, oldScore)
+		log.L().Debug("no better peer", zap.Uint64("region", region.GetID()), zap.Float64("new score", newScore), zap.Float64("old score", oldScore))
```
Comment (disksing) [inline server/schedule/replica_checker.go]: Inconsistent naming style (space separated instead of kebab)

**Comment by rleungx** [comment]
@disksing Yes. And this PR is still working in progress. Also, it needs to wait for pingcap/log#2 merging.

File: server/schedule/replica_checker.go (hunk 16 above)
Comment (rleungx) [reply server/schedule/replica_checker.go]: BTW, for some common use like `regionID` and `storeID`, I think we need to make them consistent.
/cc @nolouch

**Comment by disksing** [comment]
LGTM except for the inconsistent naming.

**Pull request merged**